evo_researcher/benchmark/benchmark.py

```diff
@@ -32,9 +32,11 @@ def __init__(
         agents: t.List[AbstractBenchmarkedAgent],
         metric_fns: t.Dict[str, t.Callable] = {},
         cache_path: t.Optional[str] = None,
+        only_cached: bool = False,
     ):
-        self.markets: t.List[Market] = markets
         self.registered_agents: t.List[AbstractBenchmarkedAgent] = agents
+        if len(set(a.agent_name for a in self.registered_agents)) != len(self.registered_agents):
+            raise ValueError("Agents must have unique names")
```

Comment:
This was a very painful bug on my side :-D

Reply:
eeek, good spot!